### DISCUSSION TITLE
How to delete documents from index and return affected rows count?

### DISCUSSION THREAD

**Discussion by d47081** [manticoresoftware/manticoresearch-php#196] [Q&A] [answered]
I found `Index::deleteDocuments` method, but don't understand how to work with.

Is any documentation or an example?
Also I want to display affected rows by this operation.

```
// src/Manticoresearch/Index.php

    public function deleteDocuments($query)
    {
        if ($query instanceof Query) {
            $query = $query->toArray();
        }
        $params = [
            'body' => [
                'index' => $this->index,
                'query' => $query
            ]
        ];
        if ($this->cluster !== null) {
            $params['body']['cluster'] = $this->cluster;
        }
        return $this->client->delete($params);
    }
```

Thanks

**Comment by d47081** [comment]
Well, here is reply:

```
    ...
    $query = new \Manticoresearch\Query();

    $query->add(
        'field',
        'condition'
    );

    $result = $index->deleteDocuments(
        $query
    );

   echo  $result['deleted'];
```

**Reply by d47081** [reply]
No, this query removes all records

I'm just looking for API solution to make same action as:
```
MySQL [(none)]> delete from yo_ygg where match('@url download');
Query OK, 327989 rows affected (0.930 sec)
```

**Comment by sanikolaev** [comment, marked as answer]
Here's the docs https://github.com/manticoresoftware/manticoresearch-php/blob/master/docs/indexclass.md#deletedocuments

Here's an example:

```
➜  ~ cat test.php
<?php
require_once __DIR__ . '/vendor/autoload.php';

$config = ['host'=>'127.0.0.1','port'=>9308];
$client = new \Manticoresearch\Client($config);
$index = $client->index('t');
$index->drop(true);
$index->create([
    'f' => ['type' => 'text'],
]);

$index->addDocuments([
  ['f' => 'a'],
  ['f' => 'b'],
  ['f' => 'a']
]);

print_r($index->deleteDocuments(['match' => ['f' => 'a']]));
➜  ~ php test.php
Array
(
    [_index] => t
    [deleted] => 2
)
```

**Reply by d47081** [reply]
Thank you much, that's it!

**Reply by d47081** [reply]
Not sure, but this construction sensitive to `min_word_len` and `min_prefix_len`.

I suppose, because any rule with single char, e.g. links with anchors (that contains single `#` char in URL), so execution by this way may also delete everything else. 

Currently, I have another implementation with the search request first, then additional check with PHP `strpos` method in loop (before delete each document by ID).

Just note for others

**Reply by sanikolaev** [reply]
It's true that when you delete documents using a full-text query, you need to double-check to ensure it deletes only what you intend. If your query consists solely of delimiters, such as `#!.-?`, you might end up deleting all documents because it could match any document.
